File: 934-bitwise-ors-of-subarrays/bitwise-ors-of-subarrays.java
import java.util.HashSet;

class Solution {
    public int subarrayBitwiseORs(int[] arr) {
        HashSet<Integer> uniqueResult = new HashSet<>();
        HashSet<Integer> previous = new HashSet<>();

        for (int i = 0; i < arr.length; i++) {
            HashSet<Integer> current = new HashSet<>();
            current.add(arr[i]);
            uniqueResult.add(arr[i]);

            for (Integer prev : previous) {
                int orResult = prev | arr[i];
                current.add(orResult);
                uniqueResult.add(orResult);
            }
            previous = current;
        }
        return uniqueResult.size();
    }
}

File: 934-bitwise-ors-of-subarrays/bitwise-ors-of-subarrays.js
var subarrayBitwiseORs = function(arr) {
    let res = new Set();
    let cur = new Set();
    for (let num of arr) {
        let next = new Set([num]);
        for (let x of cur) {
            next.add(x | num);
        }
        cur = next;
        for (let x of cur) res.add(x);
    }
    return res.size;
};